Hm 6

SunnatBakidjanov opened the pull request.

SunnatBakidjanov committed:
Добавлена логика бесконечного слайдера

--Добалвена валидация validateID в Restaurants
--Добавлена логика валидции в Restaurants
--Добавлена логика слайдера в useSlider
--Добавлена логика слайдера в TabsButtons
--Добавлена логика слайдера в RestaurantTabs
--Добавлена валидация validateResturantItem в TabsButtons

SunnatBakidjanov committed:
'Доработана логика слайдера'

--Доработана логика слайдера
--Добавлена логика слайдера в TabsButtons
--Доработана структура в RestaurantTabs

SunnatBakidjanov committed:
Кнопки prev и next выделены в отдельный компонент

--Добавлен файл SwitchButtons
--Добавлен файл switchButton.module.scss
--Логика в tabsButtons убрана в отдельные файл в папке scripts

SunnatBakidjanov committed:
'Изменения в нескольких файлах'

-----welcomePage-----

--Добавлен компонент WelcomePage.jsx + стили
--Добалвен компонент cuisinesTitle.jsx + стили
--Добавлен компонент formButtom.jsx + стили
--Добавлен компонент restaurantFilter.jsx + стили
--Перенесен компонент title(h1) + стили в отдельный компонент Title.jsx
--Добавлены отдельные стили для заголовков в scss файле
--Добавлена папка hooks

-----Общее-----

--Добавлены и подключены шрифты
--Добавлены кратинки
--Добавлен компонент Logo.jsx + стили
--Изменен Layout

-----HTML-----

--Добавлена новая иконка фавикон
--Изменен title

-----Header-----

--Добавлены стили для header

-----Общие стили-----

--Добавлены множество переменных в main.scss
--Немного изменен reset.scss

SunnatBakidjanov committed:
'Изменены несколько файлов'

----Общее----

--Перенесены шрифты и картинки в папку public

----WelcomePage----

--Создан компонент WelcomeSlides.jsx + стили
--Переделана логика слайдов в useWelcomeSlider.js

SunnatBakidjanov committed:
Удален слайдер кнопок в TabsButtons

--Удален файл useSlider
--Удалена папка scripts в restaurantTabs
--Удалена логика слайдера в restaurants

SunnatBakidjanov committed:
'Небольшие правки'

SunnatBakidjanov committed:
Merge branch 'stylization' into hm-6

SunnatBakidjanov committed:
Добавлен контекст изменения темы

--Добавлен файл contexts
--Добавлен файл theme
--Добавлен контекст themeContext.js
--Добавлен провайдер ThemeContextProvider.jsx
--Добалвены темы themes.js

-----WelcomePage-----
--Разделены компоненты слайдера и удалены не нужные
--Добавлен компонент slideContent(разделена логика слайдера)
--Добавлен компонент slideImg(разделена логика слайдера)
--Удален компонент slider
--Добавлена логика изменения темы в WelcomePage

SunnatBakidjanov committed:
Изменена логика отображения темы

--Заменены атрибуты style на классы

SunnatBakidjanov committed:
Добавлен контекст авторизации

--Добавлен файл authorization в contexts
--Добавлена логика контекста в index.js
--Добавлен провайдер AuthorizationContextProvider
--Добавлена кнопка войти/выйти
--Добалвена логика контекста
--Добавлена логика контекста в ReviewForm
--Добавлена логика контекста в Counter
--total-coast перенесен из Dish в Counter

zubkov7 commented on src/components/app/main.scss:
закомментированный код лучше удалять

zubkov7 commented on src/components/header/UI/authButton/AuthButton.jsx:
```diff
@@ -0,0 +1,25 @@
+    const { theme } = use(ThemeContext);
+
+    return (
+        <button
+            type='button'
+            onClick={toggleAuth}
+            className={classNames(styles.button, {
+                [styles.mainTheme]: theme === "main",
+                [styles.secondaryTheme]: theme === "secondary",
+            })}
+        >
```
давайте вынесем кнопку в отдельный компонент и переиспользуем в проекте

zubkov7 commented on src/components/progressBar/progressBar.module.scss:
```diff
@@ -1,14 +1,22 @@
 .container {
     position: fixed;
+    z-index: 100;
```
работать с z индексами нужно осторожно

в индексах во-первых нужно выделять верхнеуровневые слои и внутри слоя проставлять индексы 1, 2, 3 и так далее

обычно эти значения выносятся в константы

давайте здесь сделаем здесь так

SunnatBakidjanov replied:
вы имеет ввиду в :root создать я правильно понял?

SunnatBakidjanov replied:
вы имеет ввиду в :root создать я правильно понял?

zubkov7 replied:
можем на следующей лекции обсудить вопрос с z-index их нужно правильно организовать. Какая у вас сейчас проблема?

zubkov7 commented on src/components/restaurants/Restaurants.jsx:
```diff
@@ -1,22 +1,25 @@
 
 export const Restaurants = () => {
-    const { activeTabId, handleSetIdOnClick } = useTabs(restaurants);
+    const updateRestaurants = [...restaurants];
```
для чего тут создание нового массива?

SunnatBakidjanov replied:
Я думал что бы как-то не мутировать исходный массив, потому-что у меня используется в нескольких местах mock ресторана

zubkov7 replied:
так на каждый рендер у вас пересоздается массив. Для чего нужно мутировать исходный массив?

zubkov7 replied:
пересоздание осталось

zubkov7 commented on src/components/welcomePage/UI/restaurantFilter/RestaurantFilter.jsx:
```diff
@@ -0,0 +1,5 @@
+import stylesContainer from "../../scss/welcomeContainer.module.scss";
+
+export const RestaurantFilter = () => {
```
для чего используется этот компонент? css модули нужно размещать рядом с компонентом и располагать селекторы, который относятся только к данному компоненту

если селекторы переиспользуются между компонентами, то это нужно выделять в отдельный компонент и создавать к нему модуль

SunnatBakidjanov replied:
Этот компонент будет использоваться в будущем для фильтрации ресторанов просто пока выделил блок к нему, что бы у меня была разметка, хорошо я сделаю к нему отдельный модуль.

zubkov7 commented on src/components/welcomePage/UI/slideImg/WelcomeSlideImg.jsx:
```diff
@@ -0,0 +1,5 @@
+import styles from "./img.module.scss";
+
+export const WelcomeSlideImg = ({ restaurant, opacity, transform }) => {
+    return <div className={styles.img} style={{ backgroundImage: `url(/img/png/${restaurant.id}.png)`, opacity: opacity, transform: `translateX(${-transform}px)`, transition: "opacity 1s ease-out, transform 1s ease-out" }} />;
```
почему стилизация через style?

SunnatBakidjanov replied:
Это слайдер, я использую id ресторана = название изображения, логика слайдера ниже была, мне показалось так удобнее, если делать через стили тогда нужно создавать отдельный класс под каждую картинку!?

zubkov7 replied:
transition не завязан просто ни на что, его нужно вынести через класс

zubkov7 commented on src/components/welcomePage/WelcomePage.jsx:
```diff
@@ -0,0 +1,29 @@
+import titleStyles from "./scss/title.module.scss";
+
+export const WelcomePage = () => {
+    const updateRestaurants = [...restaurants];
```
для чего тут создание нового массива?

SunnatBakidjanov replied:
Аналогично выше, что бы случайно не мутировать исходный массив

zubkov7 replied:
тут тоже

zubkov7 commented on src/components/welcomePage/hooks/useWelcomeSlider.js:
```diff
@@ -0,0 +1,92 @@
+
+const START_POSITION = 0;
+
+const INITIAL_VALUE = restaurants => {
```
таким стилем именуются константные значения, функции именуются в camelCase

zubkov7 commented on src/components/welcomePage/hooks/useWelcomeSlider.js:
```diff
@@ -0,0 +1,92 @@
+const START_POSITION = 0;
+
+const INITIAL_VALUE = restaurants => {
+    const VALIDATE_POSITION = Math.max(0, Math.min(restaurants.length - 1, START_POSITION));
```
тут аналогично, в таком стиле именуются только внешние константы, переменные именуются в camelCase

zubkov7 commented on src/components/welcomePage/hooks/useWelcomeSlider.js:
```diff
@@ -0,0 +1,92 @@
+export const useWelcomeSlider = restaurants => {
+    const [state, dispatch] = useReducer(reducer, INITIAL_VALUE(restaurants));
+
+    useEffect(() => {
```
нужно чистить таймауты в cleanup функции

zubkov7 commented on src/components/welcomePage/hooks/useWelcomeSlider.js:
```diff
@@ -0,0 +1,92 @@
+                }, TIMERS.TRANSITION_DURATION);
+                break;
+
+            case 50:
```
лучше значения вынести в константы, они выше переиспользуются

zubkov7 commented on src/components/welcomePage/hooks/useWelcomeSlider.js:
```diff
@@ -0,0 +1,92 @@
+export const useWelcomeSlider = restaurants => {
+    const [state, dispatch] = useReducer(reducer, INITIAL_VALUE(restaurants));
+
+    useEffect(() => {
```
нужно добавить очистку таймаутов на анмоунт компонента, можно сделать ref для айдишников и в отдельном эффекте почистить их

SunnatBakidjanov replied:
Если честно с очисткой таймаутов у меня возникли проблемы, хорошо я попробую сделать как вы сказали

zubkov7 reviewed (commented):
поправьте работу с css модулями и проверьте пожалуйста нейминги файлов и папок, в которых они лежат. Они должны соответствовать друг другу

SunnatBakidjanov committed:
Добавлен компонент и удалнеы некоторые компоненты

--Удален закомментированный код в main.scss
--Удален шрифт Roboto
--Удален файл scss в папке header
--Удален компонент AuthButton.jsx
--Удален компонент TogleThemeButton.jsx
--Создан компонент Button.jsx

SunnatBakidjanov committed:
'Сделаны правки'

-----useWelcomeSlider.js-----

--Переименованы переменны в camelCase в
--Добавлены константы для позиционирования
--Добавлена логика сброса тайматуов
------------------------------------
Добавлена функция setSafeTimeout которая удаляет все
таймауты при анмунте компонента в useEffect

-----Общее-----
--Созданы переменные z-index в main.scss
--Разделена логика стилей по компонентам в welcomePage
--Удалена папка scss в welcomePage

zubkov7 commented on src/components/progressBar/progressBar.module.scss:
```diff
@@ -1,14 +1,22 @@
 .container {
     position: fixed;
+    z-index: var(--progress-bar);
```
тут точно нужен z-index? fixed выносится в отдельный контекст наложения, относительно чего тут идет выравнивание ?

https://developer.mozilla.org/ru/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context

zubkov7 replied:
в лекционном проекте position fixed и без z-index рисуется поверх

zubkov7 commented on src/components/restaurants/counter/Counter.jsx:
```diff
@@ -1,5 +1,11 @@
+export const Counter = ({ count, price, onIncrement, onDecrement }) => {
+    const { auth } = use(AuthContext);
+
+    return auth.isAuthed ? (
```
это логику нужно унести в потребителя, чтобы UI компонент не определял, когда должен отрисоваться

zubkov7 commented on src/components/welcomePage/UI/formButton/formButtom.module.scss:
```diff
@@ -0,0 +1,48 @@
+.after {
+    position: absolute;
+    z-index: -1;
```
для чего отрицательный индекс?

zubkov7 commented on src/components/welcomePage/UI/slideContent/WelcomeSlideContent.jsx:
```diff
@@ -0,0 +1,27 @@
+    const { theme } = use(ThemeContext);
+
+    return (
+        <div className={styles.content} style={{ opacity: opacity, transform: `translateX(${transform}px)`, transition: "opacity 1s ease-out, transform 1s ease-out" }}>
```
тут тоже transition нужно расположить в модуле

zubkov7 commented on src/components/welcomePage/hooks/useWelcomeSlider.js:
```diff
@@ -0,0 +1,112 @@
+import { useReducer, useEffect, useRef } from "react";
+
+const startPosition = 0;
```
это внешняя константа, нужно именовать в стиле как константы ниже

zubkov7 commented on src/components/welcomePage/hooks/useWelcomeSlider.js:
```diff
@@ -0,0 +1,112 @@
+
+    const timeoutsRef = useRef([]);
+
+    const setSafeTimeout = (callback, delay) => {
+        const id = setTimeout(callback, delay);
+        timeoutsRef.current.push(id);
+    };
```
нужно объявить внутри эффекта, чтобы не пересоздавать при ререндерах

zubkov7 commented on src/components/app/main.scss:
```diff
@@ -1,3 +1,65 @@
+    --welcome-page-indents: 60px;
+    --border-size: 2px;
+
+    // z-index
+    --z-index-background: 0;
+    --z-index-content: 1;
+    --z-index-progress-bar: 50;
+    --z-index-modal: 100;
+
+    // var-index
+    --progress-bar: var(--z-index-progress-bar);
```
из этого используется только прогресс бар, остальные можно удалить

SunnatBakidjanov committed:
Сделаны правки

--Изменены index в formButton.jsx
--Логика авторизации перенесна из компонентов Counter.jsx и ReviewForm.jsx
к потребителям
--Свойство transition перенесено из style в модули в WelcomeSliderContent.jsx
и WelcomeSliderImg.jsx

-----useWelcomeSlider-----
--Функци setSafeTimeout добавлена внуторь эффекта
--Вся логика добавлена в один эффект
--Изменено имя константы srtartPosition в START_POSITION

SunnatBakidjanov committed:
Убран log из RestaurantItem.jsx

zubkov7 commented on src/components/app/main.scss:
```diff
@@ -1,3 +1,61 @@
+    --border-size: 2px;
+
+    // z-index
+    --z-index-background: 1;
+    --z-index-content: 2;
+    --z-index-progress-bar: 50;
+}
```
очень сильная градация для --z-index-progress-bar можно 3

zubkov7 commented on src/components/restaurants/tabsButtons/tabsButtons.module.scss:
```diff
@@ -0,0 +1,12 @@
+.last {
+    position: absolute;
+    left: -300px;
+}
```
для чего используется?

zubkov7 reviewed (approved):
можете мержить после исправления

SunnatBakidjanov committed:
Небольшие правки
